platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py

```diff
@@ -514,7 +534,9 @@ def initialize_components(self):
             from .component import ComponentONIE, ComponentSSD, ComponentBIOS, ComponentCPLD
             self._component_list.append(ComponentONIE())
             self._component_list.append(ComponentSSD())
-            self._component_list.append(ComponentBIOS())
+            # Upgrading BIOS is not supported on SN2201
+            if DeviceDataManager.get_platform_name() not in ['x86_64-nvidia_sn2201-r0']:
+                self._component_list.append(ComponentBIOS())
```

Comment:
i suggest to add a clear message to the use in case this is asked by the user. for now it seems nothing is provided and the user will now know why it is not done